src/include/migraphx/shape.hpp

```diff
@@ -156,8 +156,33 @@ struct shape
 
     shape(const std::vector<shape>& subs);
 
+    /**
+     * Creates an output shape with dimensions equal to the input lengths and strides determined
+     * by the permutation argument.
+     * The "original" dimensions of the shape can be considered to be the forward permutation of the
+     * given output dimensions.
+     * The output shape's strides are the inverse permutation of the standard strides generated by
+     * the "original" dimensions.
```

Comment:
This is really confusing. The strides are set so that `find_permutation` returns the same permutation. There is no need to mention an inverse permutation, its not really relevant.

Reply:
I kept the inverse_perm explanation as that's what the code is doing.

Reply:
Its an implementation detail(there are other ways this could be implemented) and doesnt explain what the function is doing.